solver-next/memorycache.go

```diff
@@ -47,14 +47,78 @@ func (ck *inMemoryCacheKey) Index() Index {
 	return ck.output
 }
 
+func withExporter(ck *inMemoryCacheKey, result Result, deps [][]CacheKeyWithSelector) ExportableCacheKey {
+	return ExportableCacheKey{ck, &cacheExporter{
+		inMemoryCacheKey: ck,
+		result:           result,
+		deps:             deps,
+	}}
+}
+
+type cacheExporter struct {
+	*inMemoryCacheKey
+	result Result
+	deps   [][]CacheKeyWithSelector
+}
+
+func (ce *cacheExporter) Export(ctx context.Context, m map[digest.Digest]*ExportRecord, converter func(context.Context, Result) (*Remote, error)) (*ExportRecord, error) {
```

Comment:
I guess most of this logic is not specific to `memorycache`, but can be separated later

Reply:
correct